src/Hangfire.Core/Dashboard/Content/js/Page.js

```diff
@@ -0,0 +1,490 @@
+                    if (addedDateTimeFilterStrings || addedFilterString) redirectToFirstPage();
+                    
+                    document.location.search = url;
+                })
+                
+                var prepFilterStringParameters = function (filterString, filterMethodString) {
+                    var result = false;
+                    if (url == '' && (filterString != '' || filterMethodString != '')) {
+                        url = '?';
+                        if (filterString != '') url += "filterString=" + filterString;
+                        if (filterMethodString != '') url += "filterMethodString=" + filterMethodString;
+                        result = true;
+                    } else {
+
+                        var res1 = updateOrRemoveSingleParameter("filterString", filterString);
+                        var res2 = updateOrRemoveSingleParameter("filterMethodString", filterMethodString);
+                        var parameters = url.substr(1).split('&');
+
+                        if (!res1 && filterString != '') {
+                            parameters[parameters.length] = ["filterString", filterString].join('=');
+                            result = true;
+                        }
+
+                        if (!res2 && filterMethodString != '') {
+                            parameters[parameters.length] = ["filterMethodString", filterMethodString].join('=');  
+                            result = true;
+                        }
+                        
+                        if ( parameters.length > 0 ) url = '?' + parameters.join('&');
+                    }
+                    return result;
+                }
+
+                var updateOrRemoveSingleParameter = function (parameter, value) {
+                    var result = false;
+                    var parameters = url.substr(1).split('&');
+                    var elements;
+                    var i = 0;
+                    for (i = 0; i < parameters.length; i++) {
+                        elements = parameters[i].split('=');
+                        if (elements[0] == parameter && (typeof value == 'undefined' || value == '') ) {
+                            parameters.splice(i, 1);
+                            result = false;
+                            break;
+                        } else if (elements[0] == parameter && value != '') {
+                            elements[1] = value;
+                            parameters[i] = elements.join('=');
+                            result = true;
+                            break;
+                        } 
+                    }
+                    if (parameters.length > 0) url = '?' + parameters.join('&');
+                    return result;
+                }
+
+                var prepFilterDateTimeParameters = function (startDateTime, endDateTime) {
+                    if (typeof startDateTime == 'undefined' || typeof endDateTime == 'undefined') return false;
```

Comment:
think this should be using === and checking against undefined, rather than 'undefined' in the quotes.